Framework/Core/Windows/Devices/I2c/I2cDevice.cs

```
@@ -0,0 +1,206 @@
+                return ExecuteTransactions(transactions);
+            }
+        }
+
+        public void WriteRead(byte[] writeBuffer, byte[] readBuffer)
+        {
+            WriteReadPartial(writeBuffer, readBuffer);
+        }
+
+        public I2cTransferResult WriteReadPartial(byte[] writeBuffer, byte[] readBuffer)
+        {
+            lock (m_syncLock)
+            {
+                if (m_disposed)
+                {
+                    throw new ObjectDisposedException();
+                }
+
+                var transactions = new I2CDevice.I2CTransaction[] {
+                    I2CDevice.CreateWriteTransaction(writeBuffer),
+                    I2CDevice.CreateReadTransaction(readBuffer) };
+                return ExecuteTransactions(transactions);
+            }
+        }
+
+        public void Dispose()
+        {
+            lock (m_syncLock)
+            {
+                if (!m_disposed)
+                {
+                    Dispose(true);
+                    GC.SuppressFinalize(this);
+                    m_disposed = true;
+                }
+            }
+        }
+
+        // Private methods
+
+        private I2cTransferResult ExecuteTransactions(I2CDevice.I2CTransaction[] transactions)
+        {
+            uint bytesRequested = 0;
+            foreach (var transaction in transactions)
+            {
+                bytesRequested += (uint)transaction.Buffer.Length;
+            }
+
+            I2cTransferResult result;
+
+            lock (s_deviceLock)
+            {
+                // TODO: How long should our transaction timeout be? 15ms is the max in UWP, but too short for us.
+                s_device.Config = m_configuration;
+                result.BytesTransferred = (uint)s_device.Execute(transactions, 1000);
+            }
+
+            // TODO: We need a more reliable way to report SlaveAddressNotAcknowledged.

```

The HAL/PAL layer should report an error when a transaction is NAK'd. What's the defined UWP behavior for this?